### ISSUE TITLE
Add staff-line/IC-bbox overlays and hybrid-override labeling to demo figures

### ISSUE THREAD

**Issue by kunfang98927** [DDMAL/Standalone-Pitch-Finder#3]
demo_examples.py's illustrative crop figure didn't show what geometry a prediction was actually based on — no indication of where the detected staff lines fell within each crop, or how far the crop extended past the glyph's own IC bounding box. This made it hard to visually sanity-check a prediction or explain why a given crop looked the way it did.

This issue adds:

- staff-line overlays (magenta) and the glyph's own pre-expansion IC bbox (cyan) to every crop panel, in both demo_examples.py and the new demo_examples_all.py (same rendering, paginated across the full 247-glyph test set)
- a "[hybrid override]" tag on the uncorrected panel whenever the hybrid crop-selection rule (see #1) replaced that glyph's crop, plus a dedicated example category so the curated figure always includes a concrete override case
- reorganizes all generated figures into a figures/ subfolder, and adds visualize_many_augmentations.py for a broader augmentation sanity check

**Comment by fujinaga** [comment]
@kunfang98927 Can I see some example images of this?

**Comment by kunfang98927** [comment]
Sure.

Each row shows one glyph under two staff-line conditions: corrected staff (left, human-fixed geometry) and uncorrected staff (right, the geometry automatic staff-finding actually produced).

- Pink lines = the staff lines detected on the page.
- Blue lines = the glyph's own original ground truth bounding box (provided by Gen).

<img width="1102" height="3052" alt="Image" src="https://github.com/user-attachments/assets/897621a6-87ac-43d6-b5c6-5e7adc5e215f" />

**Comment by fujinaga** [comment]
Thank you. I want to see the entire input and output pages.

**Comment by kunfang98927** [comment]
In the following image, the input is the original page with orange bounding boxes. The output is pitch position for each neume bbox. Only bounding boxes of test set are shown in this image, and the rest of neumes are in training set. (Blue stafflines are extracted by our pretrained staff finding model and are used by the heuristic pitch finding method).

In the image, "H" means predicted by heuristic method, and "C" stands for CNN-based pitch finder. Red means wrong prediction while green means correct prediction. Stave positions are numbered bottom-to-top, starting at 0.

<img width="1800" height="2387" alt="Image" src="https://github.com/user-attachments/assets/3f5f40e2-712b-4581-b0dc-28f08674e5dc" />

**Comment by fujinaga** [comment]
I think the first thing to do is to improve staff line positions (@kyrieb-ekat, @Yueqiao12Zhang, @lizima).
